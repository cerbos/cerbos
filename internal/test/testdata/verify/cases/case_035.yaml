description: Resources filtered with default policy version configured
config:
  excludedResourcePolicyFqns:
    - cerbos.resource.leave_request.v20210210
