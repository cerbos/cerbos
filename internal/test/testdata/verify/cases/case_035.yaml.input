-- testdata/principals.yaml --
---
principals:
  john:
    id: john
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    id: XX125
    attr:
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design

  shoes_purchase_order:
    kind: purchase_order
    policyVersion: default
    id: X111
    attr:
      ownerOrgId: foo
      tags:
        brand: brand
        class: Footwear
        region: EMEA

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
options:
  defaultPolicyVersion: '20210210'
tests:
  - name: John and his leave request
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - view:public
        - approve
        - defer
      auxData: myJWT
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW
          approve: EFFECT_DENY
          defer: EFFECT_ALLOW

  - name: John and a purchase order
    input:
      principals:
        - john
      resources:
        - shoes_purchase_order
      actions:
        - view
      auxData: myJWT
    expected:
      - principal: john
        resource: shoes_purchase_order
        actions:
          view: EFFECT_DENY
