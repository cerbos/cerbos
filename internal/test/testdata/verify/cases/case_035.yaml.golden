{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 4,
        "resultCounts": [
          {
            "result": "RESULT_SKIPPED",
            "count": 3
          },
          {
            "result": "RESULT_PASSED",
            "count": 1
          }
        ]
      },
      "testCases": [
        {
          "name": "John and his leave request",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_SKIPPED",
                        "skipReason": "Resource matched a policy that was excluded from the bundle"
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_SKIPPED",
                        "skipReason": "Resource matched a policy that was excluded from the bundle"
                      }
                    },
                    {
                      "name": "defer",
                      "details": {
                        "result": "RESULT_SKIPPED",
                        "skipReason": "Resource matched a policy that was excluded from the bundle"
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "John and a purchase order",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "shoes_purchase_order",
                  "actions": [
                    {
                      "name": "view",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 4,
    "resultCounts": [
      {
        "result": "RESULT_SKIPPED",
        "count": 3
      },
      {
        "result": "RESULT_PASSED",
        "count": 1
      }
    ]
  }
}
